skglm/solvers/cd_solver.py

```diff
@@ -235,38 +230,43 @@ def cd_solver(
     stop_crit = np.inf  # initialize for case n_iter=0
 
     is_sparse = sparse.issparse(X)
+
+    if len(w) != n_features + fit_intercept:
+        raise ValueError(
+            "Shape of w should be n_features + 1 when fit_intercept=True.")
+
     for t in range(max_iter):
         if is_sparse:
             grad = datafit.full_grad_sparse(
                 X.data, X.indptr, X.indices, y, Xw)
         else:
-            grad = construct_grad(X, y, w, Xw, datafit, all_feats)
+            grad = construct_grad(X, y, w[:n_features], Xw, datafit, all_feats)
 
         if ws_strategy == "subdiff":
-            opt = penalty.subdiff_distance(w, grad, all_feats)
+            opt = penalty.subdiff_distance(w[:n_features], grad, all_feats)
         elif ws_strategy == "fixpoint":
-            opt = dist_fix_point(w, grad, datafit, penalty, all_feats)
+            opt = dist_fix_point(w[:n_features], grad, datafit, penalty, all_feats)
```

Comment:
as @Badr-MOUFAD pointed out, we don't take the intercept into account in the optimality condition. Maybe let's just write it as a comment here so that we don't forget